lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1971,28 +2000,83 @@ estimateTxSize skeleton =
         + sizeOf_Hash32
         + sizeOf_UInt
 
-    -- transaction_output =
+    -- legacy_transaction_output =
     --   [address, amount : value]
     -- value =
     --   coin / [coin,multiasset<uint>]
-    sizeOf_Output TxOut {address, tokens}
+    sizeOf_LegacyTransactionOutput TxOut {address, tokens}
         = sizeOf_SmallArray
         + sizeOf_Address address
         + sizeOf_SmallArray
         + sizeOf_Coin (TokenBundle.getCoin tokens)
         + sumVia sizeOf_NativeAsset (TokenBundle.getAssets tokens)
 
+    -- post_alonzo_transaction_output =
+    --   { 0 : address
+    --   , 1 : value
+    --   , ? 2 : datum_option ; New; datum option
+    --   , ? 3 : script_ref   ; New; script reference
+    --   }
+    -- value =
+    --   coin / [coin,multiasset<uint>]
+    sizeOf_PostAlonzoTransactionOutput TxOut {address, tokens}
+        = sizeOf_SmallMap
+        + sizeOf_SmallUInt
+        + sizeOf_Address address
+        + sizeOf_SmallUInt
+        + sizeOf_SmallArray
+        + sizeOf_Coin (TokenBundle.getCoin tokens)
+        + sumVia sizeOf_NativeAsset (TokenBundle.getAssets tokens)
+
+    sizeOf_Output
+        = case era of
```

Comment:
why not to write just 
```
          (AnyCardanoEra BabbageEra) -> sizeOf_PostAlonzoTransactionOutput
          _ -> sizeOf_LegacyTransactionOutput
```
for the sake of being more concise?

Reply:
Safety for future eras.

Ideally I would like an `Ord` instance for eras so I can express:

```
if era <= AlonzoEra
then sizeOf_LegacyTransactionOutput
else sizeOf_PostAlonzoTransactionOutput
```

But we don't have an `Ord` instance for eras :/

```
(AnyCardanoEra BabbageEra) -> sizeOf_PostAlonzoTransactionOutput 
_ -> sizeOf_LegacyTransactionOutput
```

Would mean a future era would use the `sizeOf_LegacyTransactionOutput` function. I would rather a bit of wordiness now and a compile-time warning later, than a run-time error later.

Reply:
what about this https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs#L665 ?